src/services/survey-administration-service.ts

```diff
@@ -157,11 +157,18 @@ export const close = async (id: number) => {
   if (surveyAdministration.status !== SurveyAdministrationStatus.Ongoing) {
     throw new CustomError("Only ongoing status is allowed.", 403)
   }
-
-  await SurveyAdministrationRepository.updateStatusById(
-    surveyAdministration.id,
-    SurveyAdministrationStatus.Closed
-  )
+  const currentDate = new Date()
+  const endDate = new Date(currentDate)
+
+  // Set the end time to 11:59 PM
+  endDate.setHours(23, 59, 59, 999)
+  const timeUntilEndOfDay = endDate.getTime() - currentDate.getTime()
+  setTimeout(async () => {
+    await SurveyAdministrationRepository.updateStatusById(
+      surveyAdministration.id,
+      SurveyAdministrationStatus.Closed
+    )
+  }, timeUntilEndOfDay)
```

Comment:
No need to do this, just change the survey-administration-repository from `lte` to `lt` like this

![image](https://github.com/nerubia/kh360-nodejs/assets/91468846/d0bf6d49-6a9a-475b-a2ab-bfe36977dd8b)